lnrpc/rpc.proto

```diff
@@ -1830,9 +1854,35 @@ message ChanPointShim {
     bytes pending_chan_id = 5;
 }
 
+message PsbtShim {
+    /**
+    The size of the output to be created as the channel point for this channel
+    funding.
+    */
+    int64 amt = 1;
+
+    /**
+    If non-zero, then this will be used as the pending channel ID on the wire
+    protocol to initiate the funding request. This is an optional field, and
+    should only be set if the responder is already expecting a specific pending
+    channel ID.
+    */
+    bytes pending_chan_id = 2;
+}
+
 message FundingShim {
     oneof shim {
+        /**
+        A channel shim where the channel point was fully constructed outside
+        of lnd's wallet and the transaction might already be published.
+        */
         ChanPointShim chan_point_shim = 1;
+        
+        /**
+        A channel shim that uses a PSBT to fund and sign the channel funding
```

Comment:
So the idea is that this will be specified when _initiating_ the channel opening right?

Reply:
Yes. I was confused about the use of the `shim_register` in `FundingStateStep`. See comment further below.